dhall-docs/src/Dhall/Docs/Html.hs

```diff
@@ -12,20 +12,49 @@
 {-# LANGUAGE RecordWildCards      #-}
 
 module Dhall.Docs.Html
-    ( filePathHeaderToHtml
+    ( dhallFileToHtml
     , indexToHtml
     , DocParams(..)
     ) where
 
-import Data.Monoid ((<>))
-import Data.Text   (Text)
+import Data.Monoid  ((<>))
+import Data.Text    (Text)
+import Dhall.Core   (Expr, Import)
+import Dhall.Pretty (Ann (..))
+import Dhall.Src    (Src)
 import Lucid
-import Path        (Abs, Dir, File, Path, Rel)
+import Path         (Abs, Dir, File, Path, Rel)
 
 import qualified Control.Monad
 import qualified Data.Text
+import qualified Data.Text.Prettyprint.Doc.Render.Util.SimpleDocTree as Pretty
+import qualified Dhall.Pretty
 import qualified Path
 
+exprToHtml :: Expr Src Import -> Html ()
+exprToHtml expr = renderTree prettyTree
+  where
+    prettyTree = Pretty.treeForm
+        $ Dhall.Pretty.layout
+        $ Dhall.Pretty.prettyExpr expr
+
+    textSpaces :: Int -> Text
+    textSpaces n = Data.Text.replicate n (Data.Text.singleton ' ')
+
+    renderTree :: Pretty.SimpleDocTree Ann -> Html ()
+    renderTree sds = case sds of
+        Pretty.STEmpty -> return ()
+        Pretty.STChar c -> toHtml $ Data.Text.singleton c
+        Pretty.STText _ t -> toHtml t
+        Pretty.STLine i -> br_ [] >> toHtml (textSpaces i)
+        Pretty.STAnn ann content -> encloseInTagFor ann (renderTree content)
+        Pretty.STConcat contents -> foldMap renderTree contents
+
+    encloseInTagFor :: Ann -> Html () -> Html ()
+    encloseInTagFor ann = span_ [class_ classForAnn]
+      where
+        classForAnn = "dhall-" <> Data.Text.pack (show ann)
```

Comment:
I would recommend producing the class with explicit pattern matching on the `Ann`, similar to `annToAnsiStyle`. If constructor names change or constructors are added, that might otherwise produce a bug here that could be difficult to track down.

Reply:
Yeah, I was going to say the same thing.  `show` is generally dangerous to use for anything other than interactive debugging

Reply:
Alright, I'm updating it!